src/geo/locate.js

```diff
@@ -59,7 +125,12 @@ export default async ({ address, city, region, country }) => {
     host: pelias.hosts.structured
   }
   // not in cache, fetch it
-  const out = handleQuery(opts)
+  let out
+  try {
+    out = await handleQuery(opts)
+  } catch (err) {
+    out = await locateIntersection({ intersection: address, city, region, country })
```

Comment:
Can we make this more intelligent? I don't want to send 2 requests for invalid addresses.

Here's how I think this can work:
- New util `geo.intersection`
- `geo.locate` at the very beginning will try to detect if the address is an intersection, and return `geo.intersection(options)` if it is

Gives us:
- Cleaner split of normal address code and intersection specific stuff (`src/geo/locate.js` is getting too long)
- Max 1 request per function call